source/MaterialXGenOgsFx/OgsFxShaderGenerator.cpp

```diff
@@ -238,17 +237,17 @@ ShaderPtr OgsFxShaderGenerator::generate(const string& name, ElementPtr element,
     }
     emitScopeBegin(fx);
     emitLine("pass p0", fx, false);
-    std::string vertexDataInstance = HW::T_VERTEX_DATA_INSTANCE;
-    tokenSubstitution(_tokenSubstitutions, vertexDataInstance);
     emitScopeBegin(fx);
-    emitLine("VertexShader(in VertexInputs, out VertexData " + vertexDataInstance +") = { VS }", fx);
+    emitLine("VertexShader(in VertexInputs, out VertexData " + HW::T_VERTEX_DATA_INSTANCE +") = { VS }", fx);
     emitLine(lighting ?
-        "PixelShader(in VertexData " + vertexDataInstance + ", out PixelOutput) = { LightingFunctions, PS }" :
-        "PixelShader(in VertexData " + vertexDataInstance + ", out PixelOutput) = { PS }", fx);
+        "PixelShader(in VertexData " + HW::T_VERTEX_DATA_INSTANCE + ", out PixelOutput) = { LightingFunctions, PS }" :
+        "PixelShader(in VertexData " + HW::T_VERTEX_DATA_INSTANCE + ", out PixelOutput) = { PS }", fx);
     emitScopeEnd(fx);
     emitScopeEnd(fx);
     emitLineBreak(fx);
 
+    replaceTokens(_tokenSubstitutions, fx);
```

Comment:
This was the missing call, to perform the token substitution on the final stage.